### PR TITLE
Add protocol_version in all the required places. The version cannot yet be advanced

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui-adapter/src/execution_engine.rs [hunk 1]
```diff
@@ -382,6 +382,7 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
             vec![
                 system_object_arg,
                 CallArg::Pure(bcs::to_bytes(&change_epoch.epoch).unwrap()),
+                CallArg::Pure(bcs::to_bytes(&change_epoch.protocol_version).unwrap()),
```
Comment (lxfind) [inline crates/sui-adapter/src/execution_engine.rs:385]: It's a bit strange to me that we only pass this on safe mode when we fail to advance normally but not the regular mode.

Comment (mystenmark) [reply crates/sui-adapter/src/execution_engine.rs:385]: As we discussed on slack, I'm ok with doing it in both places. Will fix.

Comment (mystenmark) [reply crates/sui-adapter/src/execution_engine.rs:385]: i'm fairly alarmed the tests passed without this.

Comment (lxfind) [reply crates/sui-adapter/src/execution_engine.rs:386]: wait, really?

**Review by lxfind — APPROVED** [review]

File: crates/sui-adapter/src/execution_engine.rs (hunk 1 above)
Comment (mystenmark) [reply crates/sui-adapter/src/execution_engine.rs:386]: as far as i can tell, yes - i wonder if we are always hitting safe mode reconfig in the tests when this happens.

**Pull request merged**